cloud-sql-proxy/cloud-sql-proxy.sh

```diff
@@ -259,12 +323,18 @@ function main() {
       else
         echo "Service hive-metastore is not enabled"
       fi
-      systemctl stop mysql
-      systemctl disable mysql
+      if ( systemctl is-enabled --quiet mysql ); then
+        systemctl stop mysql
+        systemctl disable mysql
+      fi
     fi
     install_cloud_sql_proxy
     if [[ $enable_cloud_sql_metastore = "true" ]]; then
-      configure_sql_client
+      if [[ "${MASTER_INDEX?}" == 0 ]]; then
+        configure_sql_client
+      else
+        configure_hive_warehouse_dir
```

Comment:
I think you need to run configure_hive_warehouse_dir on `-m-0` as well. If not, you should add a comment on why it isn't necessary.

Reply:
no, it's not needed, hive warehouse url will be set for master0 as part of configure_sql_client